test/versioned/kafkajs/kafka.tap.js

```diff
@@ -43,24 +43,72 @@ tap.afterEach(async (t) => {
   await t.context.producer.disconnect()
 })
 
-tap.test('stub', async (t) => {
-  const { consumer, producer, topic } = t.context
+tap.test('send records correctly', (t) => {
+  t.plan(2)
+
+  const { agent, consumer, producer, topic } = t.context
   const message = 'test message'
 
-  await consumer.subscribe({ topics: [topic], fromBeginning: true })
-  const testPromise = new Promise((resolve) => {
-    consumer.run({
-      eachMessage: async ({ message: actualMessage }) => {
-        t.equal(actualMessage.value.toString(), message)
-        resolve()
-      }
+  agent.on('transactionFinished', (tx) => {
+    const segment = tx.agent.tracer.getSegment()
+    const foundSegment = segment.children.find((s) => s.name.endsWith(topic))
```

Comment:
good start on the tests. I would use `test/versioned/amqplib/promises.tap.js` as a baseline to figure out what else to assert.